tenders/spt/spt_module_block.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2015-2018 Contributors as noted in the AUTHORS file
+ *
+ * This file is part of Solo5, a sandboxed execution environment.
+ *
+ * Permission to use, copy, modify, and/or distribute this software
+ * for any purpose with or without fee is hereby granted, provided
+ * that the above copyright notice and this permission notice appear
+ * in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
+ * CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS
+ * OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
+ * NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+ * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/*
+ * spt_module_block.c: Block device module.
+ */
+
+#define _GNU_SOURCE
+#include <assert.h>
+#include <err.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <seccomp.h>
+
+#include "spt.h"
+
+static char *diskfile;
+static int diskfd = -1;
+
+static int handle_cmdarg(char *cmdarg)
+{
+    if (strncmp("--disk=", cmdarg, 7))
+        return -1;
+    diskfile = cmdarg + 7;
+
+    return 0;
+}
+
+static int setup(struct spt *spt)
+{
+    if (diskfile == NULL)
+        return 0; /* not present */
+
+    /* set up virtual disk */
+    diskfd = open(diskfile, O_RDWR);
+    if (diskfd == -1)
+        err(1, "Could not open disk: %s", diskfile);
+
+    spt->bi->blocki.present = 1;
+    spt->bi->blocki.block_size = 512;
```

Comment:
Would it make sense to fill this with `ioctl(diskfd, BLKSSZGET, ...)`?
I don't know much about block devices, so can't tell if that's important.

Reply:
Maybe in future but then the tender needs to check if the underlying fd is backed by a device or a file, so, not right now.